### PR TITLE
Fix PCBS integration tests for Spark-3.4

### PR DESCRIPTION
**Pull request by nartal1**
This fixes https://github.com/NVIDIA/spark-rapids/issues/7216 .

Spark-3.4 added a new config flag to enable TIMESTAMP_NTZ and added that in `class ParquetToSparkSchemaConverter`
Updated the code such that Spark-3.4 shim takes that into consideration. With this fix, the PCBS tests pass in Spark-3.4.

```
========================================================= 580 passed, 81 warnings in 168.18s (0:02:48) =========================================================
Starting with OOM injection seed: 1681945885. Set env variable SPARK_RAPIDS_TEST_INJECT_OOM_SEED to override.
2023-04-19 16:11:25 INFO     Executing global initialization tasks before test launches
```

### CODE REVIEW

File: sql-plugin/src/main/330+/scala/org/apache/spark/sql/execution/datasources/parquet/ShimCurrentBatchIterator.scala [hunk 1]
```diff
@@ -71,6 +72,8 @@ class ShimCurrentBatchIterator(
   config.setBoolean(SQLConf.PARQUET_BINARY_AS_STRING.key, false)
   config.setBoolean(SQLConf.PARQUET_INT96_AS_TIMESTAMP.key, false)
   config.setBoolean(SQLConf.CASE_SENSITIVE.key, false)
+  // timestamp_NTZ flag is introduced in Spark 3.4.0
+  ParquetTimestampNTZShims.setupTimestampNTZConfig(config, conf)
```
Comment (nartal1) [inline sql-plugin/src/main/330+/scala/org/apache/spark/sql/execution/datasources/parquet/ShimCurrentBatchIterator.scala:76]: I am not certain if this is correct thing to do. This is in 330+ shim but depending on whether the Spark version is 3.3 or 3.4, it's NOOP or config is set. We already have this distinction for Spark-3.3 -[ here](https://github.com/NVIDIA/spark-rapids/blob/801a339bb3088131110076d8f3385ad9a57761f5/sql-plugin/src/main/311until340-all/scala/com/nvidia/spark/rapids/shims/ParquetTimestampNTZShims.scala#L23) and 3.4 - [here](https://github.com/NVIDIA/spark-rapids/blob/bfdecc3714ab71f76e42686d9a9d193824ae762c/sql-plugin/src/main/340%2B/scala/com/nvidia/spark/rapids/shims/ParquetTimestampNTZShims.scala#L23).
So I didn't create another object in 330until340/340+ which would call corresponding `ParquetTimestampNTZShims.setupTimestampNTZConfig` . If this code is confusing, I can split up as mentioned before.

**Comment by nartal1** [comment]
build

**Comment by nartal1** [comment]
Marking it as draft. Holding it off until Databricks-11.3 shim is in.

File: sql-plugin/src/main/330+/scala/org/apache/spark/sql/execution/datasources/parquet/ShimCurrentBatchIterator.scala (hunk 1 above)
Comment (razajafri) [reply sql-plugin/src/main/330+/scala/org/apache/spark/sql/execution/datasources/parquet/ShimCurrentBatchIterator.scala:76]: LGTM

**Review by razajafri — APPROVED** [review]

**Comment by nartal1** [comment]
build

**Comment by nartal1** [comment]
build

**Review by NVnavkumar — APPROVED** [review]

**Pull request merged**